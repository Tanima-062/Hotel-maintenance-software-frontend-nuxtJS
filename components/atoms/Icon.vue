<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
