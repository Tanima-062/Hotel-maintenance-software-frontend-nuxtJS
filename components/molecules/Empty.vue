<template>
  <div class="flex items-center justify-center flex-col h-50">
    <Icon
      :icon="icons.alert"
      color="text-bdColor"
      class="mb-1"
      :icon-width="40"
      :icon-height="40"
    />

    <p class="text-center text-black-light">
      <slot />
    </p>
  </div>
</template>
<script lang="ts">
import { mdiAlertCircleOutline } from '@mdi/js'
import { Vue, Component } from 'vue-property-decorator'

@Component({
  components: {
    Icon: () => import('~/components/atoms/Icon.vue'),
  },
})
export default class extends Vue {
  private icons = {
    alert: mdiAlertCircleOutline,
  }
}
</script>
