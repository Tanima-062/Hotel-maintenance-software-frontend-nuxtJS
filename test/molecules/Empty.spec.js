import { mount } from '@vue/test-utils'
import Empty from '@/components/molecules/Empty.vue'
import Icon from '@/components/atoms/Icon.vue'

describe('Molecules Empty Check', () => {
  test('Empty Test', () => {
    const wrapper = mount(Empty, {
      stubs: {
        Icon,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
