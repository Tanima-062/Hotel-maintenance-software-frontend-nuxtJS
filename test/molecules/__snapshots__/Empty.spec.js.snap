// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules Empty Check Empty Test 1`] = `
<div
  class="flex items-center justify-center flex-col h-50"
>
  <div
    class="mb-1"
  >
    <svg
      class="w-7 h-7 fill-current text-bdColor"
      style="width: 40px; height: 40px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M11,15H13V17H11V15M11,7H13V13H11V7M12,2C6.47,2 2,6.5 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20Z"
      />
    </svg>
  </div>
   
  <p
    class="text-center text-black-light"
  />
</div>
`;
